README.md

```diff
@@ -2,3 +2,176 @@
+Note that if you already have a sysroot with the same name inside the `sysroots` directory, it will be overwritten by the new one.
+
+If you want to use your own sysroot, instead of generating a new one, you can skip the last instruction and just place your sysroot in the `sysroots` directory. Your sysroot directory must be renamed to `raspbian` or as the specified `TARGET_ARCHITECTURE` that you passed to `env.sh`.
+
+Create a ROS2 workspace containing the sources you need
+
+    mkdir -p ~/ros2_cc_ws/src
+    cd ~/ros2_cc_ws
+    wget https://raw.githubusercontent.com/ros2/ros2/crystal/ros2.repos
+    vcs import src < ros2.repos
+
+Ignore some packages and python dependencies (note that there is a different script for each distribution you want to  cross-compile)
+
+    cd -
+    bash ignore_pkgs.sh ~/ros2_cc_ws crystal
+
+Cross-compile the workspace
+
+    bash cc_workspace.sh ~/ros2_cc_ws
+
+The provided workspace will be mounted as a volume and it will be populated with the output of the cross-compilation.
+
+#### Install
+
+Copy the `install/lib` directory from the cross-compiled SDK to your target system.
+```
+cd ~/ros2_cc_ws
+scp -r install/lib user@address:~/ros2_crystal
+```
+
+## Cross-compile ROS2 workspaces
+
+Let's go through an example showing how to cross-compile some ROS2 packages for `raspbian` board.
+
+Source the environment variables for this architecture using
+
+    source env.sh raspbian
+
+If you have followed the instructions in the previous section, you will already have a sysroot.
+
+If you need a sysroot for the architecture, create it.
+
+    bash get_sysroot.sh
+
+Note that if you want to use your own sysroot, instead of generating a new one, you can skip the last instruction and just place your sysroot in the `sysroots` directory. The name of the sysroot directory must match the name be `raspbian`.
+
+If you are creating a new sysroot or if your sysroot does not contain the ROS2 SDK, you must add it.
+You can follow the instructions in the previous section
+
+Copy the SDK into the sysroot
+
+    cp -R ~/ros2_cc_ws/install/share/* sysroots/raspbian/usr/share/
+    cp -R ~/ros2_cc_ws/install/lib/* sysroots/raspbian/usr/lib/
+    cp -R ~/ros2_cc_ws/install/include/* sysroots/raspbian/usr/include/
+
+Note that after adding contents to the sysroot, you can always revert it back to its original state by running again the `get-sysroots.sh` script.
+
+Create a ROS2 workspace, for example containing example nodes.
+
+    mkdir -p ~/my_ros2_cc_ws/src
+    git clone https://github.com/ros2/examples ~/my_ros2_cc_ws/src/examples
+
+Now you can cross-compile the workspace
+
+    bash cc_workspace.sh ~/my_ros2_cc_ws
+
+The provided workspace will be mounted as a volume and it will be populated with the output of the cross-compilation.
+
+#### Install
+
+Copy the `install/lib` directory from the cross-compiled workspace to your target system.
+
+Prefer using `rsync` to copy over files since `scp` fails randomly. For example,
```

Comment:
What do you mean by "fail randomly"? Is it because of a weak connection? You did not mentioned that earlier when doing the scp for the SDK.

Reply:
I have seen several times `scp` stalling when copying large amount of files.
On the other hand rsync seems to perform better.

I updated the description of the reason and fixed also the code snippets